pkg/bgpv1/manager/reconciler/service.go

```diff
@@ -100,6 +116,8 @@ func (r *ServiceReconciler) requiresFullReconciliation(p ReconcileParams) bool {
 	var existingSelector *slim_metav1.LabelSelector
 	if p.CurrentServer != nil && p.CurrentServer.Config != nil {
 		existingSelector = p.CurrentServer.Config.ServiceSelector
+	} else {
+		return true // the first reconciliation should be always full
```

Comment:
Do we still need this after we introduce `Init`?

Reply:
Yes, I think we should still start with a full reconcile - otherwise we could miss the events between the `DiffStore` creation and `InitDiff` call for the instance.